### PR TITLE
Modify llvm-gsymutil to ignore invalid file indexes

### PR DESCRIPTION
**Pull request by clayborg**
DWARF produced by LTO and BOLT can sometimes be broken where file indexes are beyond the end of the line table's file list in the prologue. This patch allows llvm-gsymutil to convert this DWARF without crashing, and emits errors when:

line table contains entries with an invalid file index (line entry will be removed) inline functions that have invalid DW_AT_call_file file indexes when there are no line table entries for a function and we fall back to making a single line table entry from the functions DW_AT_decl_file/DW_AT_decl_line attributes, we make sure the DW_AT_decl_file attribute is valid before emitting it.

### CODE REVIEW

**Review by kusmour — APPROVED** [review]

**Comment by dwblaikie** [comment]
> line table contains entries with an invalid file index (line entry will be removed) inline functions that have invalid DW_AT_call_file file indexes when there are no line table entries for a function and we fall back to making a single line table entry from the functions DW_AT_decl_file/DW_AT_decl_line attributes, we make sure the DW_AT_decl_file attribute is valid before emitting it.

This is hard to follow - I guess it's mean to be a bulleted list of the different cases/how they're handled, but ends up as a big run-on sentence?

**Comment by dwblaikie** [comment]
> DWARF produced by LTO and BOLT can sometimes be broken where file indexes are beyond the end of the line table's file list in the prologue. 

Bugs are always possible, but I'm pretty sure LTO would haev a hard time producing this sort of bug... - maybe BOLT's the root cause? In any case, as noted previously, some kind of bug reports would be appreciated.

**Comment by ayermolo** [comment]
> > DWARF produced by LTO and BOLT can sometimes be broken where file indexes are beyond the end of the line table's file list in the prologue.
> 
> Bugs are always possible, but I'm pretty sure LTO would haev a hard time producing this sort of bug... - maybe BOLT's the root cause? In any case, as noted previously, some kind of bug reports would be appreciated.

For once it is not. :D

So I dug into it internally a bit. Looks like issue was with llvm-12, and goes away with llvm-15 (version we are migrating to currently).
